src/main/java/lotto/domain/LottoBall.java

```diff
@@ -0,0 +1,56 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public final class LottoBall implements Comparable<LottoBall> {
+  private static final int MIN = 1;
+  private static final int MAX = 45;
+  private static final List<LottoBall> CACHE_LOTTOBALL = new ArrayList<>();
+
+  static {
+    for (int i = MIN ; i <= MAX ; i++ ) {
+      CACHE_LOTTOBALL.add(new LottoBall(i));
+    }
+  }
+
+  public static List<LottoBall> values() {
+    return CACHE_LOTTOBALL;
+  }
+
+  private final int number;
+
+  public LottoBall(final int number) {
+    if (number < MIN || number > MAX) {
+      throw new IllegalArgumentException("로또공 생성가능한 숫자가 아닙니다.");
+    }
+    this.number = number;
+  }
+
+  public int getNumber() {
+    return this.number;
+  }
+
+  @Override
+  public int compareTo(LottoBall other) {
+    return this.number > other.getNumber() ? 1 : -1;
```

Comment:
다음과 같이 변경해볼 수 있을 것 같아요 :)
추가로 같은 타입이라면 getter를 사용하지 않고 바로 내부에 접근할 수 있습니다 :)
```suggestion
    return Integer.compareTo(this.number, other.number);
```